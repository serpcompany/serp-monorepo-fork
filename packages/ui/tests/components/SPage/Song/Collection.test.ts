import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import SPageSongCollection from '../../../../components/SPage/Song/Collection.vue';
import ComponentRender from '../../../componentRender';

mockNuxtImport('useSeoMeta', () => () => {});

mockNuxtImport('useSongs', () => () => ({
  songs: [
    {
      id: 1,
      name: 'Song 1',
      slug: 'song-1',
      artists: [
        { slug: 'artist-1', credit_name: 'Artist One', join_phrase: '' }
      ]
    },
    {
      id: 2,
      name: 'Song 2',
      slug: 'song-2',
      artists: [
        { slug: 'artist-2', credit_name: 'Artist Two', join_phrase: '' }
      ]
    }
  ],
  pagination: { totalItems: 2 }
}));

mockNuxtImport('useRouter', () => () => ({
  push: () => {},
  replace: () => {},
  resolve: () => {
    return { href: '' };
  }
}));

describe('SPageSongCollection Snapshot', () => {
  const scenarios: [string][] = [['with defaults']];

  it.each(scenarios)('renders %s correctly', async (desc: string) => {
    const html = await ComponentRender(
      `SPageSongCollection ${desc}`,
      {},
      SPageSongCollection
    );
    expect(html).toMatchSnapshot();
  });
});
