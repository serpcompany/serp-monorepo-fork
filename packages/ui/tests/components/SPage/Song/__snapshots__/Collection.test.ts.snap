// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageSongCollection Snapshot > renders with defaults correctly 1`] = `
"<div class="container py-20">
  <h1 class="py-10">Songs</h1>
  <div class="mt-2 grid grid-cols-1 gap-2 md:grid-cols-2 xl:grid-cols-2">
    <div><a href="" aria-label="song name"><span>Song 1 - </span></a></div>
    <div><a href="" aria-label="song name"><span>Song 2 - </span></a></div>
  </div>
  <div class="my-20 flex w-full justify-center">
    <nav aria-label="pagination">
      <div class="flex items-center gap-1"><button type="button" disabled="" aria-label="First Page" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:chevrons-left shrink-0 size-5" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><button type="button" disabled="" aria-label="Previous Page" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:chevron-left shrink-0 size-5" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><button type="button" data-type="page" aria-label="Page 1" aria-current="page" data-selected="true" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 text-inverted bg-primary hover:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary p-1.5" value="1">
          <!--v-if--><span class="truncate min-w-5 text-center">1</span>
          <!--v-if-->
        </button><button type="button" disabled="" aria-label="Next Page" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:chevron-right shrink-0 size-5" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><button type="button" disabled="" aria-label="Last Page" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:chevrons-right shrink-0 size-5" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button></div>
    </nav>
  </div>
</div>"
`;
